### DISCUSSION TITLE
Query about data processing/location

### DISCUSSION THREAD

**Discussion by ThomasAndrewMills** [microsoft-search/pnp-modern-search#3463] [Q&A] [answered]
Hello,

This isn't really a big or a feature request so apologies. I have used PnP Modern Search whilst in a previous role and it worked great.

I want to use this again at a different organisation. However, I have a query about the data that PnP Modern Search requires to operate properly. My understanding is that once we upload the add-in into our tenancy, it keeps data local and does not send or process any data outside of the tenancy?

Can you confirm this is the case?

Thanks in advance!

**Comment by kasperbolarsen** [comment, marked as answer]
Absolutely. The web parts are only showing the data that are returned from either the SharePoint Search index or the Microsoft Search index/Graph so the data is never moved and permissions are handled by the index.

**Reply by wobba** [reply]
And we are not logging any PnP telemetry either.
